connexion/validators/json.py

```diff
@@ -52,30 +52,107 @@ def validate(self, body: dict):
             )
             raise BadRequestProblem(detail=f"{exception.message}{error_path_msg}")
 
-    @staticmethod
-    def parse(body: str) -> dict:
+    def parse(self, body: str) -> dict:
         try:
             return json.loads(body)
         except json.decoder.JSONDecodeError as e:
             raise BadRequestProblem(str(e))
 
     async def wrapped_receive(self) -> Receive:
         more_body = True
+        messages = []
         while more_body:
             message = await self._receive()
-            self._messages.append(message)
+            messages.append(message)
             more_body = message.get("more_body", False)
 
-        bytes_body = b"".join([message.get("body", b"") for message in self._messages])
+        bytes_body = b"".join([message.get("body", b"") for message in messages])
         decoded_body = bytes_body.decode(self.encoding)
 
         if decoded_body and not (self.nullable and is_null(decoded_body)):
             body = self.parse(decoded_body)
             self.validate(body)
 
         async def receive() -> t.MutableMapping[str, t.Any]:
-            while self._messages:
-                return self._messages.pop(0)
+            while messages:
+                return messages.pop(0)
+            return await self._receive()
+
+        return receive
+
+
+class DefaultsJSONRequestBodyValidator(JSONRequestBodyValidator):
+    """Request body validator for json content types which fills in default values. This Validator
+    intercepts the body, makes changes to it, and replays it for the next ASGI application."""
+
+    def __init__(self, *args, **kwargs):
+        defaults_validator = self.extend_with_set_default(Draft4RequestValidator)
+        super().__init__(*args, validator=defaults_validator, **kwargs)
+
+    # via https://python-jsonschema.readthedocs.io/
+    @staticmethod
+    def extend_with_set_default(validator_class):
+        validate_properties = validator_class.VALIDATORS["properties"]
+
+        def set_defaults(validator, properties, instance, schema):
+            for property, subschema in properties.items():
+                if "default" in subschema:
+                    instance.setdefault(property, subschema["default"])
+
+            yield from validate_properties(validator, properties, instance, schema)
+
+        return jsonschema.validators.extend(
+            validator_class, {"properties": set_defaults}
+        )
+
+    async def read_body(self) -> t.Tuple[str, int]:
+        """Read the body from the receive channel.
+
+        :return: A tuple (body, max_length) where max_length is the length of the larges message.
+        """
+        more_body = True
+        max_length = 256000
```

Comment:
Where does this number come from? I'm not aware of an ASGI limit

Reply:
This is not constrained by ASGI, but by the server. I ran some tests with Uvicorn and this was the upper limit, although I couldn't find where this limit is defined.